[BALANCE|FIX]ERT Balance Pass: 1 + Minor Ops Ship Fix

Steelpoint opened the pull request:
After some feedback from players this is the first balance pass I will be making, its a minor one though.

Features
- Combat Defibrillator: A scaled down Defib unit for ERT Med Officers, can fit in your backpack. Functions identically to the normal defib. Uses custom sprites.
- Replaced the Energy Guns in the ERT Armoury with Pulse Carbines.
- The Pulse Pistol & Carbine now use Electronic Loyalty Firing Pins, meaning only loyally implanted people can use their guns.
- ERT Officers now have all access to the station.
- Changed Pulse Carbine/Pistols to use a normal cell and not a super cell, this is to get rid of the 10 minutes recharge time. No ammo change.

---
- Fixed the Ops shuttle/base misaligned areas.
- Fixes missing firing pins for ERT Ion Rifles.

Steelpoint committed:
ERTBalancePassV1

Steelpoint commented:
@paprka may be interested in the defibrillator.

Dorsisdwarf commented:
Uh, what?
From my interactions in the playtest on the highlander round, when the ERT have pulse weapons, they will use pulse weapons, and only pulse weapons, for that 50 damage deathshot. It's not like they're going to get in trouble for pulsing people to death instead of lasering them, so why would they ever use anything but DESTROY?

Steelpoint commented:
Is this merge ready?

Steelpoint commented:
Doing some additional balance changes based off a recent, and very successful, ERT run.

Steelpoint committed:
AllAccess+EasierSuitIdentification

Steelpoint commented:
More feedback from the server.

People really seem to like the ERT, more so the admins, and they seem to do well for themselves in any given situation, from combat operations to rescue and repair. There were some additional concerns raised though.

* ERT Officers have a hard time figuring out which role they are.
* Their Guns take way too long to recharge.
* They lack access to anywhere on the station, thus forcing them to use their Pulse guns, thus making point two more obviously worse.

As such I've added two additional changes, firstly all Officers will now have All Access to the entire station (one assume Centcom would be at least that generous). Secondly I've changed the personal lockers to show a departmental sprite corresponding to the ID it holds, so a Med Officers locker will show a Med Locker sprite. I've also better highlighted the Suit Storage units.

paprka commented:
I think people are just stupid and wait for their guns to fully recharge. They recharge at the exact same rate as any other gun

Steelpoint commented:
Pulse Guns use a very high cap cell, meaning it takes around 8 minutes to fully recharge any Pulse Gun, Rifle to Pistol. 

I'll investigate Sec Uniforms later on and post some pics on the forums BEFORE I do a PR that results in 200 comments.

paprka commented:
yeah you can find the code somewhere in cheridan's pin PR if it wasn't added, he might have removed it but it's in the first commit i think

Steelpoint commented:
This is merge ready, can it be merged?

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into ERTBalanceV1

Steelpoint committed:
MergeConflict

Steelpoint commented:
20 minutes for travis to finish....

This is merge ready.

paprka commented:
sorry we need 6 weeks to review this

Steelpoint committed:
FixMissingIonGunFiringPin+LoyaltyPinsForPulseGuns

Steelpoint commented:
I fixed the missing firing pins inside the Ion Guns, and also gave all ERT Pulse weapons a loyalty firing pin.

Steelpoint commented:
Making me beg. Can we please merge this.

Miauw62 commented on code/game/objects/items/weapons/cards_ids.dm:
```diff
@@ -196,7 +196,17 @@ update_label("John Doe", "Clowny")
+	            access_hydroponics, access_library, access_lawyer, access_virology, access_cmo, access_qm, access_surgery,
+	            access_theatre, access_research, access_mining, access_mailsorting, access_weapons,
+	            access_heads_vault, access_mining_station, access_xenobiology, access_ce, access_hop, access_hos, access_RC_announce,
+	            access_keycard_auth, access_tcomsat, access_gateway, access_mineral_storeroom, access_minisat)
```
just make one ERT type and derive the different cards from that, don't copypaste all these accesses for every type.

Steelpoint committed:
AccessRefinement

Steelpoint commented:
Done, I refined the access.

RemieRichards commented on code/game/objects/items/weapons/cards_ids.dm:
```diff
@@ -196,7 +196,8 @@ update_label("John Doe", "Clowny")
 	registered_name = "Emergency Response Team Commander"
 	assignment = "Emergency Response Team Commander"
-	access = list(access_sec_doors, access_engine, access_medical)
+	New()
```
Full typepaths please. also ```access = get_all_accesses()``` wasn't quite what you were asked to do.
Miauw meant make an ```/obj/item/weapon/card/id/ert``` which has ```access = get_all_accesses()``` on it (or some other form of all access) and just have ```/obj/item/weapon/card/id/ert/Security```, ```/obj/item/weapon/card/id/ert/Engineer``` etc.

Steelpoint committed:
MoreRefinment

Steelpoint commented:
Additional refinement

Steelpoint commented:
Jesus these travis wait times.

RemieRichards commented on code/modules/projectiles/guns/energy/pulse.dm:
```diff
@@ -21,6 +21,10 @@
+	cell_type = "/obj/item/weapon/stock_parts/cell"
+
+/obj/item/weapon/gun/energy/pulse_rifle/pulse_carbine/loyalpin
+	pin = "/obj/item/device/firing_pin/implant/loyalty"
```
```/obj/item/device/firing_pin/implant/loyalty``` not ```"/obj/item/device/firing_pin/implant/loyalty"```

RemieRichards commented on code/modules/projectiles/guns/energy/pulse.dm:
```diff
@@ -29,6 +33,10 @@
+	cell_type = "/obj/item/weapon/stock_parts/cell"
+
+/obj/item/weapon/gun/energy/pulse_rifle/pulse_pistol/loyalpin
+	pin = "/obj/item/device/firing_pin/implant/loyalty"
```
```/obj/item/device/firing_pin/implant/loyalty``` not ```"/obj/item/device/firing_pin/implant/loyalty"```

Steelpoint committed:
MoreMoreRefinements

Steelpoint commented:
Travis is, again, finished.

Steelpoint commented:
This is ready to merge?!?!?
